### PR TITLE
Electron, Positron and Photon propagation

### PR DESCRIPTION
**Pull request by Jean1995**

### CODE REVIEW

File: private/PROPOSAL/crossection/PhotoPairInterpolant.cxx [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        , photoangle_(photoangle.clone()){
+    // Use own initialization
+    PhotoPairInterpolant::InitdNdxInterpolation(def);
+}
+
+
+PhotoPairInterpolant::PhotoPairInterpolant(const PhotoPairInterpolant& param)
+        : CrossSectionInterpolant(param)
+{
+    delete photoangle_;
```
Comment (Jean1995) [inline private/PROPOSAL/crossection/PhotoPairInterpolant.cxx]: delete command is wrong here

File: private/PROPOSAL/crossection/PhotoPairInterpolant.cxx [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+PhotoPairInterpolant::PhotoPairInterpolant(const PhotoPairProduction& param, const PhotoAngleDistribution& photoangle, InterpolationDef def)
+        : CrossSectionInterpolant(DynamicData::Epair, param)
+        , photoangle_(photoangle.clone()){
+    // Use own initialization
+    PhotoPairInterpolant::InitdNdxInterpolation(def);
+}
+
+
+PhotoPairInterpolant::PhotoPairInterpolant(const PhotoPairInterpolant& param)
+        : CrossSectionInterpolant(param)
```
Comment (Jean1995) [inline private/PROPOSAL/crossection/PhotoPairInterpolant.cxx]: photoangle not cloned

File: private/PROPOSAL/Propagator.cxx [hunk 3]
```diff
@@ -1348,69 +1365,70 @@ Sector::Definition Propagator::CreateSectorDefinition(const std::string& json_ob
...
+        if (json_global["compton_multiplier"].is_number())
         {
-            sec_def_global.utility_def.epair_def.lpm_effect = json_global["lpm"].get<bool>();
-            sec_def_global.utility_def.brems_def.lpm_effect = json_global["lpm"].get<bool>();
+            sec_def_global.utility_def.compton_def.multiplier = json_global["compton_multiplier"].get<double>();
         }
         else
         {
-            log_fatal("The given lpm option is not a bool.");
+            log_fatal("The given compton_multiplier option is not a double.");
```
Comment (MaxNoe) [inline private/PROPOSAL/Propagator.cxx]: See above

File: private/PROPOSAL/Propagator.cxx [hunk 4]
```diff
@@ -1348,69 +1365,70 @@ Sector::Definition Propagator::CreateSectorDefinition(const std::string& json_ob
...
         }
     }
     else
     {
-        log_debug("No given lpm option given. Use default (true)");
+        log_debug("No given compton_multiplier option given. Use default (%f)", sec_def_global.utility_def.compton_def.multiplier);
```
Comment (MaxNoe) [inline private/PROPOSAL/Propagator.cxx]: Copy paste? Additional given and using.

File: private/PROPOSAL/Propagator.cxx [hunk 5]
```diff
@@ -1348,69 +1365,70 @@ Sector::Definition Propagator::CreateSectorDefinition(const std::string& json_ob
...
-        if (json_global["mupair_enable"].is_boolean())
+        if (json_global["photopair_multiplier"].is_number())
         {
-            sec_def_global.utility_def.mupair_def.mupair_enable = json_global["mupair_enable"].get<bool>();
+            sec_def_global.utility_def.photopair_def.multiplier = json_global["photopair_multiplier"].get<double>();
         }
         else
         {
-            log_fatal("The given mupair_enable option is not a bool.");
+            log_fatal("The given photopair_multiplier option is not a double.");
```
Comment (MaxNoe) [inline private/PROPOSAL/Propagator.cxx]: se above

File: private/PROPOSAL/Propagator.cxx [hunk 6]
```diff
@@ -1348,69 +1365,70 @@ Sector::Definition Propagator::CreateSectorDefinition(const std::string& json_ob
...
         }
     }
     else
     {
-        log_debug("No given mupair_enable option given. Use default (false)");
+        log_debug("No given photopair_multiplier option given. Use default (%f)", sec_def_global.utility_def.photopair_def.multiplier);
```
Comment (MaxNoe) [inline private/PROPOSAL/Propagator.cxx]: see above

File: private/PROPOSAL/Propagator.cxx [hunk 7]
```diff
@@ -1348,69 +1365,70 @@ Sector::Definition Propagator::CreateSectorDefinition(const std::string& json_ob
...
         else
         {
-            log_fatal("The given mupair_particle_output option is not a bool.");
+            log_fatal("The given lpm option is not a bool.");
         }
     }
     else
     {
-        log_debug("No given mupair_particle_output option given. Use default (true)");
+        log_debug("No given lpm option given. Use default (true)");
```
Comment (MaxNoe) [inline private/PROPOSAL/Propagator.cxx]: see above

File: private/PROPOSAL/Propagator.cxx [hunk 8]
```diff
@@ -1348,69 +1365,70 @@ Sector::Definition Propagator::CreateSectorDefinition(const std::string& json_ob
...
         else
         {
-            log_fatal("The given weak_enable option is not a bool.");
+            log_fatal("The given mupair_particle_output option is not a bool.");
         }
     }
     else
     {
-        log_debug("No given weak_enable option given. Use default (false)");
+        log_debug("No given mupair_particle_output option given. Use default (true)");
```
Comment (MaxNoe) [inline private/PROPOSAL/Propagator.cxx]: see above

File: private/PROPOSAL/Propagator.cxx [hunk 9]
```diff
@@ -1531,6 +1549,24 @@ Sector::Definition Propagator::CreateSectorDefinition(const std::string& json_ob
...
+            sec_def_global.utility_def.ioniz_def.parametrization = IonizationFactory::Get().GetEnumFromString(ioniz);
+        }
+        else
+        {
+            log_fatal("The given ioniz option is not a string.");
+        }
+    }
+    else
+    {
+        log_debug("The ioniz option is not set. Use default %s",
```
Comment (MaxNoe) [inline private/PROPOSAL/Propagator.cxx]: see above

File: private/PROPOSAL/Propagator.cxx [hunk 10]
```diff
@@ -1567,6 +1603,24 @@ Sector::Definition Propagator::CreateSectorDefinition(const std::string& json_ob
...
+            sec_def_global.utility_def.annihilation_def.parametrization = AnnihilationFactory::Get().GetEnumFromString(annihilation);
+        }
+        else
+        {
+            log_fatal("The given annihilation option is not a string.");
+        }
+    }
+    else
+    {
+        log_debug("The annihilation option is not set. Use default %s",
```
Comment (MaxNoe) [inline private/PROPOSAL/Propagator.cxx]: see above

**Pull request merged**